nemo/collections/llm/gpt/model/deepseek.py

```diff
@@ -0,0 +1,553 @@
+
+        with no_init_weights(True):
+            # Since DeepSeek is not importable from transformers, we can only initialize the HF model
+            # from a known checkpoint. The model_name will need to be passed in.
+            return AutoModelForCausalLM.from_pretrained(
+                model_name,
+                trust_remote_code=True,
+                torch_dtype=dtype,
+            )
+
+    def ckpt_load(self, path: Path) -> Tuple[Dict, Dict]:
+        """
+        This function loads the state dict directly from a distributed checkpoint, and modify the state dict
+        so that it is consistent with the key names you would get from loading the checkpoint into a model.
+        This is a more memory-efficient method to obtain a state dict without initializing the nemo model.
+
+        Args:
+            path (Path): The path from which the model will be loaded.
+
+        Returns
+        -------
+            Tuple[Dict, Dict]: The loaded state dict and the yaml config dict.
+        """
+        model_yaml = path / "context" / "model.yaml"
+        if not model_yaml.exists():
+            raise FileNotFoundError("model.yaml is not found in the context folder of the checkpoint.")
+        with open(model_yaml, 'r') as stream:
+            config = yaml.safe_load(stream)
+
+        dist_ckpt_folder = path / "weights"
+        state_dict = {}
+        for k, v in load_distributed_model_weights(dist_ckpt_folder, True).items():
+            if '_extra_state' in k:
+                continue
+            new_k = k.replace("module.", "")
+            if '.experts.experts.' in k:
+                # split experts into multiple tensors
+                for i in range(v.size(0)):
+                    state_dict[new_k.replace(".experts.experts.", ".experts.") + str(i)] = v[i]
+            else:
+                state_dict[new_k] = v
+        return state_dict, config['config']
+
+    def apply(self, output_path: Path, target_model_name=None) -> Path:
+        logging.info("Loading DeepSeek NeMo checkpoint. This may take a while...")
+        source, source_config = self.ckpt_load(self)
+        logging.info("DeepSeek NeMo checkpoint loaded.")
+        if target_model_name is None:
+            # Before DeepSeek is fully supported by HF, it is necessary to pass in a local HF checkpoint that
+            # is used to initialize the HF model. The following
+            logging.warning(
+                "Before DeepSeek is officially supported in HF, you should pass in a local HF "
+                "checkpoint using llm.export_ckpt(..., target_model_name=<local hf path>)"
+            )
+            if source_config['moe_router_enable_expert_bias']:
+                target_model_name = "deepseek-ai/DeepSeek-V3"
+            elif source_config['q_lora_rank'] is not None:
+                target_model_name = "deepseek-ai/DeepSeek-V2"
+            else:
+                target_model_name = "deepseek-ai/DeepSeek-V2-Lite"
```

Comment:
Maybe adding an additional logging of the inferred `target_model_name` when not provided explicitly for clarity ?